controllers/socialevent/socialevent_controller.go

```diff
@@ -0,0 +1,87 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// Reconciler reconciles a SocialEvent object
+type Reconciler struct {
+	client.Client
+	StatusUpdater *StatusUpdater
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&toolchainv1alpha1.SocialEvent{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Complete(r)
+}
+
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=socialevents,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=socialevents/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=socialevents/finalizers,verbs=update
+
+// Reconcile takes care of:
+// - checking that the target tier specifiec in the SocialEvent is valid and set the status condition accordingly
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	// fetch the SocialEvent resource
+	event := &toolchainv1alpha1.SocialEvent{}
+	if err := r.Client.Get(context.TODO(), request.NamespacedName, event); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("SocialEvent not found")
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		logger.Error(err, "unable to get the current SocialEvent")
+		return reconcile.Result{}, errs.Wrap(err, "unable to get the current NSTeSocialEventplateTier")
+	}
+
+	if err := r.checkTier(logger, event); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) checkTier(logger logr.Logger, event *toolchainv1alpha1.SocialEvent) error {
+	// fetch the NSTemplateTier tier specified in the SocialEvent
+	tier := &toolchainv1alpha1.NSTemplateTier{}
+	if err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: event.Namespace,
+		Name:      event.Spec.Tier,
+	}, tier); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("NSTemplateTier not found", "nstemplatetier_name", event.Spec.Tier)
+			return r.StatusUpdater.updateStatusConditions(event, toolchainv1alpha1.Condition{
+				Type:    toolchainv1alpha1.ConditionReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  toolchainv1alpha1.SocialEventInvalidTierReason,
+				Message: fmt.Sprintf("NSTemplateTier '%s' not found", event.Spec.Tier),
+			})
+		}
+		// Error reading the object - requeue the request.
+		logger.Error(err, "unable to get the NSTemplateTier", "nstemplatetier_name", event.Spec.Tier)
+		return errs.Wrapf(err, "unable to get the '%s' NSTemplateTier", event.Spec.Tier)
```

Comment:
Why don't we set the "failed" condition for this error as well?

Reply:
because we're going to requeue the request, like we do for other resources in the same case

Reply:
done in https://github.com/codeready-toolchain/host-operator/pull/624/commits/7b901ca9244cef0deb97c853d2ea1b9d8ec6756f: the status will be updated with the new `UnableToGetTier` reason introduced in [api#311](https://github.com/codeready-toolchain/api/pull/311/files#diff-21b047e72efa2ef881ad05b5b46f4fa54d832b1d43040f5282632b99527bf400R11), and the error will be returned by the controller, which should trigger another reconcile loop (as it was the case before)